scripts/robotnix_common.py

```diff
@@ -31,10 +31,10 @@ def save(filename: str, data: Any) -> None:
 
 def get_store_path(path):
     """Get actual path to a Nix store path; supports handling local remotes"""
-    prefix = os.getenv('NIX_REMOTE');
+    prefix = os.getenv('NIX_REMOTE')
     if prefix and not prefix.startswith('/'):
         raise Exception('Must be run on a local Nix store.')
-    return f"{prefix}/{path}"
+    return os.path.join(*filter(None, [prefix, path]))
```

Comment:
Sorry about that.

I'd prefer @eyJhb's patch from here: https://github.com/nix-community/robotnix/pull/228/files#diff-d2cdd059d29643c5fe1767c6dcf5203c0dc9b382d1322e3f0a4237eda6066246 though.

Reply:
Hmm, can you explain why? I've done it like this, so this function doesn't inadvertently turn a relative path to an absolute one rooted at `/`. Is doing that a desired behaviour of this function?

Reply:
This function's purpose is to turn a nix store path to an absolute path that can be opened on the filesystem.

I've rewritten it, could you take a look?

Reply:
Seems reasonable; I'll re-test it locally just in case.

Reply:
Ok, after testing I'm kind of confused on what exactly `NIX_REMOTE` is. Nix itself seems to have that parameter (https://nix.dev/manual/nix/2.22/command-ref/env-common#env-NIX_REMOTE) but it doesn't seem to be that (because it's supposed to be a daemon socket).

From the context it looks like a mountpoint for some alternate store — but also not quite. When I added some logging to the code I get paths like `/nix/store/deadbeef-hudson`, and if you relativise them to root, you'll end up with paths like `/your/prefix/nix/store/deadbeef-hudson` which makes it not a store path IMO, I'd rather expect `/your/prefix/deadbeef-hudson` in this case, as if you specified `NIX_STORE_DIR` (https://nix.dev/manual/nix/2.22/command-ref/env-common#env-NIX_STORE_DIR) with that value.

But this is a bit of a documentation issue, I guess. I proposed a bit improved implementation in a separate commit — it leaves the `path` untouched if `NIX_REMOTE` is not set, otherwise it does the same logic but with `pathlib` which makes it a bit cleaner, as you don't have to worry about `.` not being resolved or knowing what is the root path to relativise the store path against.

Reply:
`NIX_REMOTE` specifies a "remote" nix store. It's the equivalent of the `--store` flag.

If you set it to a local path, nix would then use the nix store under that prefix but it does not change the `NIX_STORE_DIR`. Build sandboxes are mounted such that the paths are available under `/nix/`.  
You can't directly run software off of such a store without a chroot but that's of little relevance for my robotnix builds.
The reason I use this is to store the 60GB+ build-time closure and hundreds of prefetched source dirs in a separate Nix store to my system's because I don't want to trash it. I explicitly do not want the `NIX_STORE_DIR` to change here.

(You can also point it at another machine via ssh in which case it builds on the remote machine but that doesn't work for the updater.) 

In the updater, we need to be able to access the `lineage.dependencies` file in a fetched repo and to do that while NIX_REMOTE points at some other location on the filesystem, you need this function. The returned string is not a store path but an absolute path that you can simply open and read.

Thanks for the pathlib patch, that looks a lot more robust. I don't have any experience with python to speak of, so this is much appreciated.

Reply:
Yeah, I did roughly understand what this function is for and it's a reasonable thing to want — I just wasn't sure what exactly it represents in terms of nix config/options/envvars (since nothing seemed to naively match it) and as such what I should expect the correct behaviour to be when testing it.

But it makes a lot more sense now that you explained what it's supposed to represent in terms of nix options/behaviour and I guess it actually indeed is `NIX_REMOTE` as described [here](https://nix.dev/manual/nix/2.22/command-ref/env-common#env-NIX_REMOTE), it's just that the description in the manual doesn't really make it clear that all values from [this list](https://nix.dev/manual/nix/2.22/command-ref/new-cli/nix3-help-stores.html) are valid and your explanation clarified that.